eventio/event_io_file.py

```diff
@@ -0,0 +1,65 @@
+import struct
+from .object_header import make_ObjectHeader
+from io import BytesIO
+
+class EventIOObject:
+    ''' A generic EventIOObject
+
+    It has a list of `headers` and a binary string `payload`.
+    The payload might be loaded lazily on first access or already in memory.
+    This is decided on construction time.
+    '''
+    def __init__(self, headers, payload=None, file=None):
+        if file is not None:
+            self._file = file
+        elif payload is not None:
+            self._file = None
+            self.payload = payload
+        else:
+            raise Exception('EventIOObject must have either file or payload')
+        
+        self.headers = headers
+
+    def __getattr__(self, attr):
+        if attr == "payload":
+            if self._file.closed:
+                self._file = open(self._file.name, 'rb')
```

Comment:
Right, any idea how to solve it?
The file could be closed before anybody ever tries to access an EventIOObject.payload.
Then we get an exception here.